File: python/day1_2.py
import sys
from collections import Counter

left = []
right =[]

for line in sys.stdin:
    a,b = [int(i) for i in line.split()]
    left.append(a)
    right.append(b)

left.sort()
right.sort()

r = Counter(right)
s=0
for i in left:
    if i in r:
        s+=i*r[i]

print(s)

File: kotlin/day1_2.kt
fun main() {
    val input = generateSequence(::readLine)
    val left = mutableListOf<Int>()
    val right = mutableListOf<Int>()

    for (line in input) {
        val ints = line.split("\\s+".toRegex()).map { it.toInt() }
        left.add(ints[0])
        right.add(ints[1])
    }
    left.sort()
    right.sort()

    val r = right.groupingBy { it }.eachCount()
    var s = 0
    for (l in left) {
        if (r.containsKey(l)) {
            s += l * r[l]!!
        }
    }
    print(s)
}
